serve_mux.go

```diff
@@ -102,7 +102,9 @@ func (mux *ServeMux) ServeDNS(w ResponseWriter, req *Msg) {
 	if h != nil {
 		h.ServeDNS(w, req)
 	} else {
-		HandleFailed(w, req)
+		m := new(Msg)
```

Comment:
this is the *only* spot where HandleFailed is actually used (internally).

I think it's better to add a handleRefused (not exported) that's called here. Potentially we can remove HandleFailed, but that would be a breaking change. We could mark it deprecated though